.github/pull_request_template.md

```diff
@@ -0,0 +1,22 @@
+# Description
+Briefly describe the problem or reason for the changes implemented in the PR
+
+# Trac ticket
+Link to the accepted Trac ticket for this PR. All PRs except docs or tests changes need an accepted ticket. New features need community consensus.
+
+# Tests
+Briefly describe any tests you have added or updated to test the changes introduced by the PR
+
+# Checklist
+- [ ] I have updated the Trac ticket’s flags
+- [ ] My commit message is written in past tense, mentions the ticket #, and ends with a dot
+- [ ] My PR targets the `main` branch. _Backport to other stable branches will be done by mergers when necessary._
+- [ ] I have added or updated tests to test the changes I made
+- [ ] I have added or updated relevant documentation (if applicable)
+
+# More details
```

Comment:
I'm not sure this section is a good idea, tbh, for multiple reasons: 

- When you open a PR, GitHub shows you a sidebar linking to the contributing.rst, which in turn links [to the contribution docs](https://docs.djangoproject.com/en/dev/internals/contributing/).
- The "Working with git" documentation is probably not the most relevant once people are already about to open a PR
- The PR template is opened in an editable text box, so using Markdown format is cumbersome, as you either have to manually select the link, or switch to the preview panel to click it
- IME, the template should be as short as possible (so people don't just delete it entirely), and only contain things that should end up in the PR text – not least of all because you can search PRs, and leaving this section in would potentially clutter up the search.

Reply:
> I'm not sure this section is a good idea, tbh, for multiple reasons:
> 
>     * When you open a PR, GitHub shows you a sidebar linking to the contributing.rst, which in turn links [to the contribution docs](https://docs.djangoproject.com/en/dev/internals/contributing/).

While this link is helpful, I think it too broad and unspecific about the stage the author is at this point.

>     * The "Working with git" documentation is probably not the most relevant once people are already about to open a PR

Good point about the git docs, though there are a few sections at the bottom of that page that are very relevant for when proposing a PR and then following up during reviews. It describes rebasing and other relevant processes. I'm proposing that we tailor the link to these sections instead.

>     * The PR template is opened in an editable text box, so using Markdown format is cumbersome, as you either have to manually select the link, or switch to the preview panel to click it
> 
>     * IME, the template should be as short as possible (so people don't just delete it entirely), and only contain things that should end up in the PR text – not least of all because you can search PRs, and leaving this section in would potentially clutter up the search.

We have many contributors not following the process of having a ticket, adding tests, adding docs, setting the trac flags for a PR. I believe that they are not following the expected procedure because they don't know about it (not because they ignore it). And I think that they don't know about it because they got lost in the contributing docs (I've been there!). More so, I think that while is likely that (new?) contributors are familiar with GH, they are not with other systems or may be discouraged by the need to jump between the docs site, the trac site, and this site.

With this information, do you think there is something we could do to improve the contributor experience and, consequently, help reviewers receive more complete PRs?

Reply:
> The PR template is opened in an editable text box, so using Markdown format is cumbersome, as you either have to manually select the link, or switch to the preview panel to click it

I think I agree with @rixx in that this information might be better placed elsewhere, since having doc links in a PR template can make the template overcrowded with info, and the fact that even opening the links from the template is a bit cumbersome is a good point.

I agree that it's easy to get lost in the contributing docs, but maybe that's something we need to fix separately?